signing/mcl/singlesig/bls_test.go

```diff
@@ -244,3 +248,81 @@ func TestBLSSigner_IsInterfaceNil(t *testing.T) {
 
 	require.False(t, check.IfNil(llSig))
 }
+
+func TestBLSSigner_TestVectorsSign(t *testing.T) {
+	t.Parallel()
+
+	jsonFile, err := os.Open("./testData/SingleSignTestVectorsSign.json")
+	require.Nil(t, err)
+	defer jsonFile.Close()
+
+	var testVar TestVectors
+	jsonDec := json.NewDecoder(jsonFile)
+	err = jsonDec.Decode(&testVar)
+	require.Nil(t, err)
+
+	signer := singlesig.NewBlsSigner()
+	suite := mcl.NewSuiteBLS12()
+	kg := signing.NewKeyGenerator(suite)
+
+	for _, testVector := range testVar.TestVectors {
+		if len(testVector.TestName) == 0 {
+			continue
```

Comment:
the dummy testname should be used only in case the testvector.TestName is empty, otherwise it should use the given testname.  Also remove the continue.

also for the other testvectors.